### PR TITLE
Batchnorm2d fixed running_var

### PR DESCRIPTION
**Pull request by symlons**
fixed running_var of BatchNorm2d where a test did not match pytorch's running_var.

BatchNorm2d is now implemented in the same way as pytorch, using the biased variance estimator for training and the unbiased estimator for inference.

Note that technically it would be more ideal to use the unbiased estimator for both training and inference, as mentioned here by Andrej Karpathy: 
https://www.youtube.com/watch?v=q8SA3rM6ckI&t=3920s

### CODE REVIEW

File: tinygrad/nn/__init__.py [hunk 1]
```diff
@@ -24,7 +25,7 @@ def __call__(self, x:Tensor):
       # NOTE: wow, this is done all throughout training in most PyTorch models
       if self.track_running_stats:
         self.running_mean.assign((1 - self.momentum) * self.running_mean + self.momentum * batch_mean.detach())
-        self.running_var.assign((1 - self.momentum) * self.running_var + self.momentum * batch_var.detach())
+        self.running_var.assign((1 - self.momentum) * self.running_var + self.momentum * prod(y.shape)/(prod(y.shape) - y.shape[1]) * batch_var.detach() )
```
Comment (geohot) [inline tinygrad/nn/__init__.py:28]: Hmm, are you sure this is right in general? You are subtracting just one of the channel axis from the product. Can you add a comment explaining this?

**Comment by geohot** [comment]
Cool! I don't exactly understand why this is right though, but I like that the test passes. See comment above.

File: tinygrad/nn/__init__.py (hunk 1 above)
Comment (symlons) [reply tinygrad/nn/__init__.py:28]: Batchnorm2D is calculated per-dimension over the mini-batches, therefore you need to subtract when using Besel's correction not just 1 but the channel axis. In test_nn.py it is tested with a channel size of 4, but I tested it also with other values and it passed those as well.

Comment (geohot) [reply tinygrad/nn/__init__.py:28]: Comment + add those tests and I think it's good.

**Comment by geohot** [comment]
Actually wait. This is in the wrong place probably. It should be on batch_var itself.

**Comment by symlons** [comment]
If it is on batch_var itself, then it won't match pytorch anymore, since pytorch uses the biased estimate for training but the unbiased one for inference.

**Comment by geohot** [comment]
Ahh, I heard this once. Crazy stuff.
